### DISCUSSION TITLE
Is it possible to define things like getters/factory methods on generated data classes?

### DISCUSSION THREAD

**Discussion by galacticgibbon** [simolus3/drift#1799] [General]
Not much else to add :) Just would be nice to be able to add custom logic to Data classes

**Comment by simolus3** [comment]
You can use [existing data classes](https://drift.simonbinder.eu/docs/advanced-features/custom_row_classes/) to make drift use a class you provide as a row class. That class can contain custom logic, but you'll need to write your own `copyWith` method if you need one.

Somewhat more advanced, you can also tell drift to let the generated data class _extend_  a class that you wrote with the [`extending` property](https://drift.simonbinder.eu/api/drift/dataclassname/extending) on `@DataClassName`. That way you just put your logic in an abstract class and drift-generated classes will inherit that logic.
